File: problems/remove_repeating_integers/remove_repeating_integers.java
class Solution {
    public static int countWithRemovedDuplicates(int[] numbers) {
      int uniqueCount = 0;
      int counter = 1;

      if (numbers.length ==0 ) {
        return 0;
      }
        
      while(counter < numbers.length) {

          if (numbers[uniqueCount] != numbers[counter]) {
            uniqueCount++;
            numbers[uniqueCount] = numbers[counter];
          }

        counter++;
      }
      return uniqueCount + 1;
  }


  public static void main(String[] args) {
    int[] numbers = {1,3,3,10};

    System.out.println(countWithRemovedDuplicates(numbers));
    System.out.println(Arrays.toString(numbers));
  }
}

File: problems/remove_repeating_integers/remove_repeating_integers.swift
func countWithRemovedDuplicates(_ numbers: inout [Int]) -> Int {
    var previousValue: Int?
    var count: Int = 0

    for value in numbers {
        if value != previousValue {
            numbers[count] = value

            previousValue = value
            count += 1
        }
    }
    return count
}

var nums = [0,0,0,1,1,2,3,3,4,4,4]
countWithRemovedDuplicates(&nums)
print(nums)
